torch/_dynamo/eval_frame.py

```diff
@@ -698,7 +705,11 @@ def result_capturing_wrapper(*graph_inputs):
     ):
         opt_f = optimize_assert(
             dynamo_normalization_capturing_compiler,
-            hooks=Hooks(guard_export_fn=guard_export_print, guard_fail_fn=None),
+            hooks=Hooks(
+                guard_export_fn=guard_export_print,
+                guard_fail_fn=None,
+                expose_shape_env_fn=expose_shape_env_fn,
+            ),
```

Comment:
You don't need this hook, I think you can just get it off the tracing context?

Reply:
Hmm tracing context is inside OutputGraph(fx.Tracer) which is not available in eval_frame.py where we only see the graph. I think I would have to somehow add the env as attribute in graph to access in export. So i thought this was cleaner.

Reply:
It is globally available! Try torch._guards.TracingContext.get()?

Reply:
ah gotcha thanks for the pointer